tuf/api/metadata.py

```diff
@@ -1371,10 +1365,7 @@ def from_data(
                     digest_object = sslib_hash.digest_fileobject(
                         data, algorithm
                     )
-            except (
-                sslib_exceptions.UnsupportedAlgorithmError,
-                sslib_exceptions.FormatError,
-            ) as e:
+            except (sslib_exceptions.FormatError,) as e:
                 raise exceptions.UnsupportedAlgorithmError(
                     f"Unsupported algorithm '{algorithm}'"
                 ) from e
```

Comment:
I mentioned this elsewhere already but after looking at this for a while I've convinced myself that we could just handle both of them and return something generic like UnsignedMetadataError like the verify functions do, no? then I think we we can remove UnsupportedAlgorithmError altogether.

Reply:
ah no, it is still used by TargetFile.from_data()